src/dragonfs.c

```diff
@@ -555,13 +572,15 @@ static int __dfs_init(uint32_t base_fs_loc)
     directory_entry_t id_node;
     grab_sector((void *)base_fs_loc, &id_node);
 
-    if(id_node.flags == ROOT_FLAGS && id_node.next_entry == ROOT_NEXT_ENTRY && 
-        !strcmp(id_node.path, ROOT_PATH))
+    if(id_node.flags == ROOT_FLAGS && !strcmp(id_node.path, ROOT_PATH))
     {
         /* Passes, set up the FS */
         base_ptr = base_fs_loc;
         clear_directory();
-
+        if(id_node.file_pointer == 0) {
+            return DFS_EBADFS;
+        }
+        init_dfs_lookup(&id_node);
```

Comment:
Can you maybe move this check into init_dfs_lookup and make it return a bool, at least it's clear that the check is related to the lookup.